src/database/database.ts

```diff
@@ -1,233 +1,47 @@
-import { URL } from 'url';
-import * as path from 'path';
-
-import { FirebaseApp } from '../firebase-app';
-import { FirebaseDatabaseError, AppErrorCodes, FirebaseAppError } from '../utils/error';
-import { FirebaseServiceInterface, FirebaseServiceInternalsInterface } from '../firebase-service';
-import { Database } from '@firebase/database';
-
-import * as validator from '../utils/validator';
-import { AuthorizedHttpClient, HttpRequestConfig, HttpError } from '../utils/api-request';
-import { getSdkVersion } from '../utils/index';
-
-
-/**
- * Internals of a Database instance.
+/*!
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  */
-class DatabaseInternals implements FirebaseServiceInternalsInterface {
-
-  public databases: {
-    [dbUrl: string]: Database;
-  } = {};
 
-  /**
-   * Deletes the service and its associated resources.
-   *
-   * @return {Promise<()>} An empty Promise that will be fulfilled when the service is deleted.
-   */
-  public delete(): Promise<void> {
-    for (const dbUrl of Object.keys(this.databases)) {
-      const db: Database = this.databases[dbUrl];
-      db.INTERNAL.delete();
-    }
-    return Promise.resolve(undefined);
-  }
-}
-
-declare module '@firebase/database' {
-  interface Database {
+// Required to perform module augmentation to FirebaseDatabase interface.
+import '@firebase/database';
```

Comment:
I couldn't fully reason about why this works.

If I do `import '@firebase/database-types'` then `npm test` fails:

```
Error: Cannot find module '@firebase/database-types'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
// ...
```

If I do no imports here, then `npm run integration` fails:
```
[10:41:29] Using gulpfile ~/Desktop/Programming-Projects/google-internship/firebase-admin-node/gulpfile.js
[10:41:29] Starting 'build'...
[10:41:29] Starting 'cleanup'...
[10:41:29] Finished 'cleanup' after 33 ms
[10:41:29] Starting 'compile'...
/Users/horatiulazu/Desktop/Programming-Projects/google-internship/firebase-admin-node/node_modules/@firebase/database/dist/index.d.ts(25,41): error TS2339: Property 'enableLogging' does not exist on type 'typeof import("@firebase/database-types")'.
/Users/horatiulazu/Desktop/Programming-Projects/google-internship/firebase-admin-node/node_modules/@firebase/database/dist/index.d.ts(26,32): error TS2694: Namespace '"@firebase/database-types"' has no exported member 'ServerValue'.
/Users/horatiulazu/Desktop/Programming-Projects/google-internship/firebase-admin-node/node_modules/@firebase/database/dist/index.d.ts(27,36): error TS2339: Property 'FirebaseDatabase' does not exist on type 'typeof import("@firebase/database-types")'.
src/database/index.ts(41,53): error TS2339: Property 'DataSnapshot' does not exist on type 'typeof import("@firebase/database-types")'.
src/database/index.ts(41,53): error TS2694: Namespace '"@firebase/database-types"' has no exported member 'DataSnapshot'.
src/database/index.ts(42,53): error TS2339: Property 'OnDisconnect' does not exist on type 'typeof import("@firebase/database-types")'.
src/database/index.ts(42,53): error TS2694: Namespace '"@firebase/database-types"' has no exported member 'OnDisconnect'.
// ...
```

When importing `@firebase/database-types`, I am not sure why this doesn't work.  

When performing no imports, my suspicion is that the module augmentation / declaration merging isn't working as expected and instead actually using it as a definition the `FirebaseDatabase` type (as-per the errors reported).

Reply:
Discussed offline. Related fix in development at https://github.com/firebase/firebase-js-sdk/pull/3511